aurora-locker/src/AuroraSdk.sol

```diff
@@ -0,0 +1,177 @@
+
+    /// Compute implicity Aurora Address for the given NEAR account.
+    function implicitAuroraAddress(string memory accountId) public pure returns (address) {
+        // TODO: Verify this returns the last 20 bytes
+        return address(bytes20(keccak256(bytes(accountId))));
+    }
+
+    /// Compute the implicit Aurora address of the represenative NEAR account
+    /// for the given Aurora address. Useful when a contract wants to call
+    /// itself via a callback using cross contract call precompile.
+    function nearRepresentitiveImplicitAddress(address account) public returns (address) {
+        return implicitAuroraAddress(nearRepresentative(account));
+    }
+
+    /// Get the promise result at the specified index.
+    function promiseResult(uint256 index) public returns (PromiseResult memory) {
+        (bool success, bytes memory returnData) = CURRENT_ACCOUNT_ID_PRECOMPILE.call("");
+        require(success);
+
+        Borsh.Data memory borsh = Borsh.from(returnData);
+
+        uint32 length = borsh.decodeU32();
+        require(index < length, "Index out of bounds");
+
+        for (uint256 i = 0; i < index; i++) {
+            borsh.skipBytes();
+        }
+
+        return borsh.decodePromiseResult();
+    }
+
+    /// Get the NEAR account id of the current contract. It is the account id of Aurora engine.
+    function currentAccountId() public returns (string memory) {
+        (bool success, bytes memory returnData) = CURRENT_ACCOUNT_ID_PRECOMPILE.call("");
+        require(success);
+        return string(returnData);
+    }
+
+    /// Get the NEAR account id of the predecessor contract.
+    function predecessorAccountId() public returns (string memory) {
+        (bool success, bytes memory returnData) = PREDECESSOR_ACCOUNT_ID_PRECOMPILE.call("");
+        require(success);
+        return string(returnData);
+    }
+
+    /// Crease a base promise. This is not immediately schedule for execution
+    /// until transact is called. It can be combined with other promises using
+    /// `then` combinator.
+    ///
+    /// Input is not checekd during promise creation. If it is invalid, the
+    /// transaction will be scheduled either way, but it will fail during execution.
+    function call(
+        NEAR memory near,
+        string memory targetAccountId,
+        string memory method,
+        bytes memory args,
+        uint128 nearBalance,
+        uint64 nearGas
+    ) public returns (PromiseCreateArgs memory) {
+        if (!near.initialized) {
+            /// If the contract needs to be initialized, we need to attach
+            /// 2 NEAR (= 2 * 10^24 yoctoNEAR) to the promise.
+            nearBalance += 2_000_000_000_000_000_000_000_000;
+            near.initialized = true;
+        }
+
+        if (nearBalance > 0) {
+            near.wNEAR.transferFrom(msg.sender, address(this), uint256(nearBalance));
+        }
+
+        return PromiseCreateArgs(targetAccountId, method, args, nearBalance, nearGas);
```

Comment:
On second thought I think the logic for `nearBalance` is wrong. The balance attached to the promise should not include the 2 NEAR to be paid in the first time.